apps/files/src/store/actions.js

```diff
@@ -93,6 +95,43 @@ function _buildFileInTrashbin (file) {
   })
 }
 
+function _buildLink (l) {
+  const link = l.shareInfo
+  let description = ''
+
+  switch (link.permissions) {
+    case ('1'):
+      description = 'Viewer' // hover: Recipients can view or download contents.
```

Comment:
translation needed?

Reply:
Do you think we should import `client` and `gettext` instead of always passing it in via a param? … or is there a deeper thought behind that?

Reply:
client and gettext are explicit instances - import will not help

we did not yet find a better solution besides adding them as arguments :man_shrugging: